grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.controller.response;
+
+import ai.grakn.exception.GraknBackendException;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * <p>
+ *     Factory used to build the wrapper {@link Concept}s from real {@link ai.grakn.concept.Concept}s
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+public class ConceptBuilder {
+
+    /**
+     * Takes a {@link ai.grakn.concept.Concept} and returns the equivalent response object
+     *
+     * @param concept The {@link ai.grakn.concept.Concept} to be converted into a response object
+     * @return the response object wrapper {@link Concept}
+     */
+    public static <X extends Concept> Optional<X> build(ai.grakn.concept.Concept concept){
+        if(concept == null) return Optional.empty();
```

Comment:
This looks like the only case where it returns `Optional.empty()`. I think it's clearer to handle it at the call-site:

```
repr = Optional.ofNullable(concept).map(ConceptBuilder::build);
```

But if you want to keep the null check, remember to mark the argument as `@Nullable` 👍